### PR TITLE
Repo cleanup pretag

### PR DESCRIPTION
**Pull request by nkern**
A few minor changes addressing issues in `UVPSpec` and `PSpecData` before we tag a release

### CODE REVIEW

**Comment by nkern** [comment]
still working on one last edit...

File: hera_pspec/pspecdata.py [hunk 1]
```diff
@@ -2075,15 +2076,43 @@ def pspec(self, bls1, bls2, dsets, pols, n_dlys=None, input_data_weight='identit
...
-                        # in this case, all Gv are the same, so skip if already built for this spw!
-                        pass
+                    if input_data_weight == 'identity':
+                        # in this case, all Gv and Hv differ only by flagging pattern
+                        # so check if we've already computed this
+                        # First: get flag weighting matrices given key1 & key2
+                        Y = np.vstack([self.Y(key1).diagonal(), self.Y(key2).diagonal()])
+
+                        # Second: check cache for Y
+                        matches = [np.isclose(Y, y).all() for y in self._identity_Y.values()]
```
Comment (philbull) [inline hera_pspec/pspecdata.py]: This is probably fine, but an alternative would be to hash the array using md5sum.

File: hera_pspec/uvpspec_utils.py [hunk 2]
```diff
@@ -241,6 +242,10 @@ def _blpair_to_antnums(blpair):
     ant2 = int(np.floor(blpair / 1e6 - ant1*1e3))
     ant3 = int(np.floor(blpair / 1e3 - ant1*1e6 - ant2*1e3))
     ant4 = int(np.floor(blpair - ant1*1e9 - ant2*1e6 - ant3*1e3))
+    ant1 -= 100
```
Comment (philbull) [inline hera_pspec/uvpspec_utils.py]: Why are these on separate lines? They could just be put into the lines above.

File: hera_pspec/pspecdata.py [hunk 3]
```diff
@@ -2075,15 +2076,43 @@ def pspec(self, bls1, bls2, dsets, pols, n_dlys=None, input_data_weight='identit
...
+                            match = self._identity_Y.keys()[matches.index(True)]
+                            Gv = self._identity_G[match]
+                            Hv = self._identity_H[match]
+                        else:
+                            # This Y doesn't exist, so compute it
+                            if verbose: print("  Building G...")
+                            Gv = self.get_G(key1, key2)
+                            Hv = self.get_H(key1, key2, sampling=sampling)
+                            # cache it
+                            self._identity_Y[(key1, key2)] = Y
```
Comment (philbull) [inline hera_pspec/pspecdata.py]: Just for clarity: This is done per baseline-pair, and not per blpair-time? So the flags still have to remain time-independent?

File: hera_pspec/uvpspec_utils.py (hunk 2 above)
Comment (nkern) [reply hera_pspec/uvpspec_utils.py]: I don't think so, see how ant2 is dependent on ant1 and so on..

File: hera_pspec/pspecdata.py (hunk 3 above)
Comment (nkern) [reply hera_pspec/pspecdata.py]: Yeah, done per-baselinepair (not time). flags still need to remain time-independent

File: hera_pspec/uvpspec_utils.py (hunk 2 above)
Comment (philbull) [reply hera_pspec/uvpspec_utils.py]: Oops, my bad.

**Pull request merged**